src/redux/books/booksSlice.js

```diff
@@ -1,43 +1,85 @@
-import { createSlice } from '@reduxjs/toolkit';
+/* eslint-disable no-param-reassign */
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+const url = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/gRb0vwPao4X8lS6fzQWO/books/';
 
 const initialState = {
-  books: [
-    {
-      item_id: 'item1',
-      title: 'The Great Gatsby',
-      author: 'John Smith',
-      category: 'Fiction',
-    },
-    {
-      item_id: 'item2',
-      title: 'Anna Karenina',
-      author: 'Leo Tolstoy',
-      category: 'Fiction',
-    },
-    {
-      item_id: 'item3',
-      title: 'The Selfish Gene',
-      author: 'Richard Dawkins',
-      category: 'Nonfiction',
-    },
-  ],
+  books: {},
+  isLoading: false,
+  error: null,
 };
 
+const fetchBooks = createAsyncThunk(
+  'books/fetchBooks',
+  async () => {
+    try {
+      const { data } = await axios.get(url);
+      return data;
+    } catch (error) {
+      return error.message;
+    }
+  },
+);
+
+const postBook = createAsyncThunk('books/addBooks', async (book) => {
+  const data = JSON.stringify(book);
+  try {
+    const response = await axios.post(url, data);
+    return response.data;
+  } catch (error) {
+    return error;
+  }
```

Comment:
- [ ] The api returns an error when the endpoint to add a new book is hit. From the error, it appears as though the `id`, `title` and `author` are not being supplied as you can see below 👇. Please look into this 👍 
![image](https://user-images.githubusercontent.com/46839548/222881708-ccd5804f-4247-427b-85df-3d1f03dcabf8.png)